Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs

```diff
@@ -354,61 +414,85 @@ private static string GetFullTypeNameWithNamespace(ITypeSymbol typeSymbol, Seman
+            if (includeAssembly)
+            {
+                qualifiedNamespace += $", {action.AssemblyName}";
+            }
+
+            return qualifiedNamespace;
+        }
+        
+        private static string GetStateKey(RGStatesInfo state, bool includeAssembly)
+        {
+            var qualifiedNamespace = "";
+            if (!string.IsNullOrEmpty(state.Namespace))
+            {
+                qualifiedNamespace += $"{state.Namespace}.";
+            }
+            qualifiedNamespace += state.Object;
+            if (includeAssembly)
+            {
+                qualifiedNamespace += $", {state.AssemblyName}";
+            }
+
+            return qualifiedNamespace;
+        }
+        
         private static void ExtractObjectType()
         {
             // read current JSON for actions and states
-            string actionJson = ReadFromJson("RGActions");
-            string stateJson = ReadFromJson("RGStates");
+            var actionJson = ReadFromJson("RGActions");
+            var stateJson = ReadFromJson("RGStates");
             
             // deserialize into action and state objects
-            RGActionsInfo actionsInfo = JsonUtility.FromJson<RGActionsInfo>(actionJson);
+            var actionsInfo = JsonUtility.FromJson<RGActionsInfo>(actionJson);
             var statesInfo = JsonConvert.DeserializeObject<RGStateInfoWrapper>(stateJson).RGStateInfo;
 
-            // all unique object names in actions and states
-            List<string> objectTypeNames = new List<string>();
-            
-            // map of object names and object types
-            Dictionary<Type, string> objectTypeMap = new Dictionary<Type, string>();
-            Dictionary<string, string> objectNameMap = new Dictionary<string, string>();
+            // map of action/state script to the entity this script is attached to
+            var objectTypeMap = new Dictionary<Type, string>();
+            var objectNameMap = new Dictionary<string, string>();
             
-            // Get all Object names from RGActions
             foreach (var action in actionsInfo.BotActions)
             {
-                string objectName = action.Object;
-                if (!objectTypeNames.Contains(objectName))
+                var key = GetActionKey(action, true);
+                var objectType = Type.GetType(key);
+                if (objectType != null)
                 {
-                    objectTypeNames.Add(objectName);
-                }
-            }
-
-            // Get all Object names from RGStates
-            foreach (var state in statesInfo)
-            {
-                string objectName = state.Object;
-                if (!objectTypeNames.Contains(objectName))
+                    objectTypeMap.TryAdd(objectType, null);
+                }else
                 {
-                    objectTypeNames.Add(objectName);
+                    RGDebug.LogWarning("Type not found: " + key);
```

Comment:
Seeing this in the log as a developer using our SDK wouldn't tell me much. 

Can we log anything else to help them identify the gameObject they're looking for ?

Reply:
This key is the assembly + full namespace + className of the script we're attempting to look up, which imo is useful information for figuring out _what_ failed, although there's not really anything a user could do to fix this on their end :-/ 
Our current approach is to look up all of the scripts and then associate the name of a GameObject to each of them - unfortunately, this step comes before we attempt to locate any gameObjects from the scene + prefabs, so we have no specific gameObject in context at this point.

I do have a task open [here](https://linear.app/regression-games/issue/REG-1375/actions-and-states-in-game-context-can-be-associated-with-multiple) for allowing multiple objectTypes to be associated to a given action/state script (currently, only one objectType "wins" and makes it into the JSON per script, which is wrong). I'm thinking when we address this task, we'll probably want to reverse this whole process - look up the gameObjects and then associate all of their known action/state scripts to them. We could probably give a more helpful warning about which gameObjects (or at least objectTypes) are affected at that point.

Reply:
Ah, you beat me to a response